client/components/contact/ContactForm.jsx

```diff
@@ -0,0 +1,270 @@
+
+  // mapStateToProps equivalent
+  const displayFeedbackSuccess = useSelector(state => state.ui.displayFeedbackSuccess);
+  const openErrorModal = useSelector(state => state.ui.error.isOpen);
+
+  const [formValues, setFormValues] = useState(initialFormValues);
+
+  function clearFields() {
+    setFormValues({
+      ...initialFormValues,
+    });
+  }
+
+  // initialize component
+  useEffect(() => {
+    // componentDidMount code goes here...
+    clearFields();
+    if (!!displayFeedbackSuccess === true) {
+      toast.success('We received your message. Our team will contact you at the email address provided.', {
+        position: 'top-right',
+        autoClose: 5000,
+        hideProgressBar: false,
+        closeOnClick: true,
+        pauseOnHover: true,
+        draggable: true,
+        progress: undefined,
+      });
+    }
+
+    if (!!openErrorModal === true) {
+      toast.error('We failed to process your message. Please try again later.', {
+        position: 'top-right',
+        autoClose: 5000,
+        hideProgressBar: false,
+        closeOnClick: true,
+        pauseOnHover: true,
+        draggable: true,
+        progress: undefined,
+      });
+    }
+
+    return () => {
+      // componentWillUnmount code goes here...
+      callShowFeedbackSuccess(false);
+      callShowErrorModal(false);
+      clearFields();
+    };
+  }, [callShowErrorModal, callShowFeedbackSuccess, displayFeedbackSuccess, openErrorModal]);
+
+  // helper methods
+  function validateEmail(emailAddress) {
+    // eslint-disable-next-line
+    if (/^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/.test(emailAddress)) {
+      return true;
+    }
+    return false;
+  }
+
+  const clearErrors = useCallback(() => {
+    setFormValues(prevState => ({
+      ...prevState,
+      ...{
+        errors: {
+          missingFirstName: false,
+          missingLastName: false,
+          missingEmail: false,
+          invalidEmail: false,
+          missingMessage: false,
+        },
+      },
+    }));
+  }, []);
+
+  const validateForm = useCallback(() => {
+    const noFirstName = formValues.firstName.trim().length === 0;
+    const noLastName = formValues.lastName.trim().length === 0;
+    const noEmail = formValues.email.trim().length === 0;
+    const noMessage = formValues.message.trim().length === 0;
+    const incompleteEmail = (!noEmail && !validateEmail(formValues.email));
+    if (!noFirstName && !noLastName && !noEmail && !noMessage && !incompleteEmail) {
+      return true;
+    }
+
+    setFormValues(prevState => ({
+      ...prevState,
+      ...{
+        errors: {
+          missingFirstName: noFirstName,
+          missingLastName: noLastName,
+          missingEmail: noEmail,
+          invalidEmail: incompleteEmail,
+          missingMessage: noMessage,
+        },
+      },
+    }));
+    return false;
+  }, [formValues]);
+
+  // event handlers
+  const onInputChange = useCallback(event => {
+    event.preventDefault();
```

Comment:
For my understanding, can you explain what this line does?

Reply:
i didn't want the default onchange event behavior to trigger. all that is required is updating the state and notice the `value` property in each TextField. the value assigned to the `value` property, which comes from the formValues state variable, gets updated whenever an onchange event happens. any time a state variable gets changed, it triggers a render() call and updates the form. disabling default onchange event behavior just prevents any potential side effects/conflicts with this process